src/hooks/useGridDimensions.ts

```diff
@@ -29,17 +34,24 @@ export function useGridDimensions(): [
 
     const resizeObserver = new ResizeObserver((entries) => {
       const size = entries[0].contentBoxSize[0];
-      setInlineSize(handleDevicePixelRatio(size.inlineSize));
+      const newInlineSize = handleDevicePixelRatio(size.inlineSize);
+      setInlineSize(newInlineSize);
       setBlockSize(size.blockSize);
+      if (prevInlineSize.current !== newInlineSize) {
+        prevInlineSize.current = newInlineSize;
+        // clear existing flex widths, this will trigger recalculation of visible flex columns again
+        setFlexColumnWidths((widths) => (widths.size > 0 ? new Map() : widths));
+      }
```

Comment:
Let's wrap this in a `useLatestFunc` so we don't re-create/disconnect the resize observer more than once.